components/Auth/SignIn.js

```diff
@@ -52,12 +53,17 @@ class SignIn extends Component {
   constructor (props) {
     super(props)
     this.state = {
-      email: props.email || '',
+      email: this.ensureDecodedEmail(props.email) || '',
       polling: false,
       loading: false,
       success: undefined
     }
   }
+
+  ensureDecodedEmail (email) {
```

Comment:
this function does not need to be on the class, you could do a small helper before the class:
```js
const ensureDecodedEmail = (email = '') => match(email) ? decode(email) : email
```
_the default argument value of `''` also makes the undefined checks and default via `or` unnecessary_